FLT/AutomorphicRepresentation/Example.lean

```diff
@@ -321,9 +321,48 @@ noncomputable abbrev zHatsub : AddSubgroup QHat :=
 noncomputable abbrev zsub : AddSubgroup QHat :=
   (Int.castRingHom QHat : ℤ →+ QHat).range
 
-lemma rat_meet_zHat : ratsub ⊓ zHatsub = zsub := sorry
+lemma rat_meet_zHat : ratsub ⊓ zHatsub = zsub := by
+  apply le_antisymm
+  · intro x ⟨⟨l, hl⟩, ⟨r, hr⟩⟩
+    simp at hl hr; rcases lowestTerms x with ⟨⟨N, z, hNz, hx⟩, unique⟩
+    have cop1 : IsCoprime l.den.toPNat' l.num := by
+      have : PNat.val l.den.toPNat' = l.den := by simp [l.den_pos]
+      simp [IsCoprime]; rw [this]
+      have hlp := (ZMod.isUnit_iff_coprime l.num.natAbs l.den).2 l.reduced
+      let m := l.num; have hm : l.num = m := by rfl
+      rcases m with m | m
+      · simp [hm] at *; exact hlp
+      · simp [hm] at *; rw [← neg_add, add_comm]; exact IsUnit.neg hlp
+    have cop2 : IsCoprime 1 r := ⟨default, by simp [ZMod.instUnique.uniq]⟩
```

Comment:
I think it's best to avoid using instance names explicitly. This could be:
```suggestion
    have cop2 : IsCoprime 1 r := ⟨default, by simp only [PNat.val_ofNat]; exact Subsingleton.elim _ _⟩
```
or also
```suggestion
    have cop2 : IsCoprime 1 r := by
      simp only [IsCoprime, PNat.val_ofNat]
      exact isUnit_of_subsingleton _
```